django_mongodb_backend/routers.py

```diff
@@ -9,10 +11,32 @@ def allow_migrate(self, db, app_label, model_name=None, **hints):
         EmbeddedModels don't have their own collection and must be ignored by
         dumpdata.
         """
+        from django_mongodb_backend.models import EmbeddedModel  # noqa: PLC0415
+
         if not model_name:
             return None
         try:
             model = apps.get_model(app_label, model_name)
         except LookupError:
             return None
+
         return False if issubclass(model, EmbeddedModel) else None
+
+
+def kms_provider(self, model, *args, **kwargs):
+    for router in self.routers:
+        func = getattr(router, "kms_provider", None)
+        if func and callable(func):
+            result = func(model, *args, **kwargs)
+            if result is not None:
+                return result
+    if model_has_encrypted_fields(model):
```

Comment:
I think `kms_provider()` should never be called for an unencrypted model so this `model_has_encrypted_fields()` check seems unnecessary.